### PR TITLE
User can choose not to delete users

### PR DESCRIPTION
**Pull request by Sesshoumaru404**
**Link back to Jira ticket:**

## Description

Allow for script to run without deleting users, helpful if changing a user to a stakeholder

## Implementation

User the `-n` flag to no delete user but remove from teams, schedules and EP

### Steps to test changes
*

## Documentation 
- [x] There is documentation that needs to be updated upon merging these changes

Links to any documentation to be updated:

### CODE REVIEW

File: user_deprovision/user_deprovision.py [hunk 1]
```diff
@@ -446,7 +453,10 @@ def delete_user(access_token, user_email, from_email, prompt_del, auto_resolve,
                 '%s:%s/{id}%s' % (method, resource, suffix), 0
             ) for method in ('put', 'delete')
         ]))
-    if user_deleter.delete_user():
+    if no_delete:
+        log.info('User %s not removed; "Do not delete user" flag selcted.', user_email)
```
Comment (JosefGoodyear) [inline user_deprovision/user_deprovision.py:457]: "selcted" > "selected"  typo

File: user_deprovision/user_deprovision.py [hunk 2]
```diff
@@ -546,6 +555,11 @@ def main(arguments):
              "directory.",
         default=False, action='store_true'
     )
+    parser.add_argument(
+        '--do-not-delete-users', '-n',
+        help="Do not delete user but preform all other actions",
```
Comment (JosefGoodyear) [inline user_deprovision/user_deprovision.py:560]: "preform" > "perform" typo

**Review by JosefGoodyear — CHANGES_REQUESTED** [review]
I feel like the initial message "{} users to be deleted" will be confusing if they aren't deleting users. Could we add the following line (or something similar) in main:

        if arguments.do_not_delete is True:
            print("{} users to be removed from schedules, teams, and escalation policies. Users will not be deleted."
            .format(len(email_list)))
        else:
            print("{} users to be deleted".format(len(email_list)))

**Review by ehlerorngard — APPROVED** [review]
This is great, Paul!  Thanks for doing it.  Works great.  I pushed up Josef's logging edits plus another couple log updates.

**Comment by ethanmsl** [comment]
Working through very old PR backlog.
I've just merged current master into this branch.
@Sesshoumaru404 -- could you re-rest the script, check that Josef's request is addressed and that you're into Ehler's adds (I assume; just catching us all up on what's happened here) then ping me and I'll review.

If time has worn down memories of this script hit me up and we'll bang through it together.

**Pull request merged**